### PR TITLE
LocalBearTestHelper.py: Allow to pass bear setting

### PR DESCRIPTION
**Pull request by tulphoon**
### Check list

- [x] I have rebased properly.
      [Please see this for a tutorial.](http://coala.readthedocs.io/en/latest/Developers/Git_Basics.html#rebasing)
- [x] I have gone through the
      [commit guidelines](http://coala.readthedocs.io/en/latest/Developers/Writing_Good_Commits.html)
      and I've followed them:
 - [x] I have followed the
       [guidelines for the commit shortlog](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#shortlog).
  - [x] I have included the issue URL with the  `Fixes` keyword if the
       commit fixes a *real bug* and the `Closes` keyword if it adds a
       feature/enhancement.
       [See more here.](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#issue-reference)
- [x] I have followed the
       [guidelines for the commit body](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#commit-body).
- [x] I have [run all the tests](http://coala.readthedocs.io/en/latest/Developers/Executing_Tests.html)
      and they all pass. If any CI (below) is not green, please fix them. If
      GitMate issues appear you will have to amend your commits, pushing new
      commits on top is not sufficient!

### Reviewers

Closes https://github.com/coala/coala/issues/2952

### CODE REVIEW

File: coalib/testing/LocalBearTestHelper.py [hunk 1]
```diff
@@ -113,7 +117,7 @@ def check_results(self,
                               list,
                               msg="The given results are not a list.")
 
-        with prepare_file(lines, filename,
+        with prepare_file(lines, filename, settings,
```
Comment (Makman2) [inline coalib/testing/LocalBearTestHelper.py]: settings should go into `execute_bear`, this is the wrong function :) This one prepares a temporary file with the contents given by `filename`, so has nothing to do with bear execution^^

File: coalib/testing/LocalBearTestHelper.py [hunk 2]
```diff
@@ -117,11 +117,11 @@ def check_results(self,
                               list,
                               msg="The given results are not a list.")
 
-        with prepare_file(lines, filename, settings,
+        with prepare_file(lines, filename,
                           force_linebreaks=force_linebreaks,
                           create_tempfile=create_tempfile,
                           tempfile_kwargs=tempfile_kwargs) as (file, fname), \
-                execute_bear(local_bear, fname, file) as bear_output:
+                execute_bear(local_bear, fname, file, settings) as bear_output:
```
Comment (Makman2) [inline coalib/testing/LocalBearTestHelper.py]: ah sorry for not mentioning, I think you need to unpack settings:
```
execute_bear(..., **settings)
```
look at the `execute_bear` definition on how to use it :)

**Comment by Makman2** [comment]
cross tested: https://github.com/coala/coala-bears/pull/978

**Comment by Makman2** [comment]
Please squash all commits together, we don't want to have fixup commits in history, as the histroy then looks like "Introduce feature A" -> "Ah, forgot to add B to A" -> "Damn, typo during B" -> etc :)

File: coalib/testing/LocalBearTestHelper.py [hunk 3]
```diff
@@ -84,6 +84,7 @@ def check_results(self,
                       local_bear,
```
Comment (Nosferatul) [inline coalib/testing/LocalBearTestHelper.py]: `Fixes` -> `Closes` in the commit message

File: coalib/testing/LocalBearTestHelper.py [hunk 4]
```diff
@@ -117,7 +121,8 @@ def check_results(self,
                           force_linebreaks=force_linebreaks,
                           create_tempfile=create_tempfile,
                           tempfile_kwargs=tempfile_kwargs) as (file, fname), \
-                execute_bear(local_bear, fname, file) as bear_output:
+                execute_bear(local_bear, fname, file,
+                        **settings) as bear_output:
```
Comment (Nosferatul) [inline coalib/testing/LocalBearTestHelper.py]: You need to align this as PEP8 says.

File: coalib/testing/LocalBearTestHelper.py (hunk 3 above)
Comment (Nosferatul) [inline coalib/testing/LocalBearTestHelper.py]: Also rebase it

File: coalib/testing/LocalBearTestHelper.py [hunk 5]
```diff
@@ -84,6 +84,7 @@ def check_results(self,
                       local_bear,
                       lines,
                       results,
+                      settings={},
```
Comment (Makman2) [inline coalib/testing/LocalBearTestHelper.py]: could you please put the parameter at the end of the parameters list? then this doesn't break current tests on coala-bears which rely on the position of each argument^^

**Comment by Makman2** [comment]
unack 8d4b338

**Comment by sils** [comment]
neat, I see forward to merging this after it's all in the right order and rebased :) (at some point you'll do a rebase more or less automatically, then it stops being so annoying ;))

File: coalib/testing/LocalBearTestHelper.py (hunk 5 above)
Comment (tulphoon) [inline coalib/testing/LocalBearTestHelper.py]: In the function, the :param settings: or both of them?

Comment (Makman2) [inline coalib/testing/LocalBearTestHelper.py]: both :+1:

File: coalib/testing/LocalBearTestHelper.py [hunk 6]
```diff
@@ -97,6 +98,9 @@ def check_results(self,
         :param lines:            The lines to check. (List of strings)
         :param results:          The expected list of results.
         :param filename:         The filename, if it matters.
+        :param settings:         A dictionary of keys and values (both string)
```
Comment (Adrianzatreanu) [inline coalib/testing/LocalBearTestHelper.py]: both strings**

File: coalib/testing/LocalBearTestHelper.py [hunk 7]
```diff
@@ -149,15 +154,15 @@ def verify_local_bear(bear,
...
-                             available for the tested bear.
     :param force_linebreaks: Whether to append newlines at each line
                              if needed. (Bears expect a \\n for every line)
     :param create_tempfile:  Whether to save lines in tempfile if needed.
     :param timeout:          The total time to run the test for.
     :param tempfile_kwargs:  Kwargs passed to tempfile.mkstemp() if tempfile
                              needs to be created.
+    :param settings:         A dictionary of keys and values (both string) from
+                             which settings will be created that will be made
+                             available for the tested bear.
```
Comment (Makman2) [inline coalib/testing/LocalBearTestHelper.py]: The change for this function does belong to a second commit, though I wouldn't change this at all^^

File: coalib/testing/LocalBearTestHelper.py (hunk 5 above)
Comment (Makman2) [inline coalib/testing/LocalBearTestHelper.py]: this one really has to be the last parameter so we don't break compatability ! :+1:

**Comment by Makman2** [comment]
unack ad32e54

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/2962/commits/a088311d3d21bd0afb6691c9b75f0b558cf54c0f

**Comment by sils** [comment]
@rultor merge

**Comment by sils** [comment]
Hey, this is your first contribution, right? Congrats and welcome to the coalaians! :)

You can help us a lot by filling out <http://coala.io/newform>!

Rultor will now automatically release your changes as a prerelease so users can experience the goodness of your changes right now.

See also https://github.com/coala-analyzer/coala/wiki/coala-Steam-Offer btw.

We would really appreciate it if you could help us with

- some code reviews for other newcomers
- fixing at least a difficulty/low issue

We're all volunteers and we're currently struggling to keep this up. Helping all those newcomers is a lot of work and we do need your help so we can continue this for others!

**Pull request merged**